tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -138,6 +138,7 @@ internal ImmutableTracerSettings(TracerSettings settings, bool unusedParamNotToU
             IsActivityListenerEnabled = settings.IsActivityListenerEnabled;
 
             _isDataStreamsMonitoringEnabled = settings.IsDataStreamsMonitoringEnabled;
+            IsDataStreamsLegacyHeadersEnabled = settings.IsDataStreamsLegacyHeadersEnabled;
```

Comment:
The lack of adding this originally feels like it should have been caught by a test 🤔 Feels like we're either missing some generalised testing, or something in this PR?